sherpa/fit.py

```diff
@@ -1052,6 +1045,16 @@ def __init__(self, data, model, stat=None, method=None, estmethod=None,
                                 itermethod_opts)
         NoNewAttributesAfterInit.__init__(self)
 
+    def calc_thaw_indices(self):
+        iter = 0
```

Comment:
This routine can be simplified to

    self.thaw_indices = tuple([i for i, par in enumerate(self.model.pars) if not par.frozen])

(not tested, I should say), which I find a bit easier to follow.

Reply:
Oh, I see I'd essentially given this as a workaround in https://github.com/sherpa/sherpa/issues/342#issuecomment-284405268

(but dropping the intermediate array in the comprehension - i.e.

    ... = tuple(i for i, ...)